### PR TITLE
Removes the small perf tool

### PR DESCRIPTION
**Pull request by derobins**
This tool doesn't really do anything special and is installed, which
conflicts with gnu's perf tool.

Fix for #1787

### CODE REVIEW

File: release_docs/RELEASE.txt [hunk 1]
```diff
@@ -139,6 +139,13 @@ New Features
 
       (ADB - 2022/06/22)
 
+    - The perf tool has been removed
```
Comment (byrnHDF) [inline release_docs/RELEASE.txt:135]: This should goto the top of the section.

File: tools/src/h5perf/CMakeLists.txt [hunk 2]
```diff
@@ -23,41 +22,14 @@ set_global_variable (HDF5_UTILS_TO_EXPORT "${HDF5_UTILS_TO_EXPORT};h5perf_serial
...
-
-    #-----------------------------------------------------------------------------
-    # Add Target to clang-format
-    #-----------------------------------------------------------------------------
-    if (HDF5_ENABLE_FORMATTERS)
-      clang_format (HDF5_TOOLS_SRC_H5PERF_perf_FORMAT perf)
-    endif ()
-  endif ()
-
   #-- Adding test for h5perf
```
Comment (byrnHDF) [inline tools/src/h5perf/CMakeLists.txt]: Might as well remove this since we are building programs - not adding tests.

Comment (derobins) [reply tools/src/h5perf/CMakeLists.txt]: Fixed

File: release_docs/RELEASE.txt (hunk 1 above)
Comment (derobins) [reply release_docs/RELEASE.txt:135]: Fixed

**Review by byrnHDF — APPROVED** [review]

**Review by lrknox — APPROVED** [review]

**Pull request merged**